03.bowling/bowling.rb

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+scores = ARGV[0].split(',')
+shots = scores.map { |score| score == 'X' ? 10 : score.to_i }
+
+frames = []
+current_frame = []
+
+shots.each do |shot|
+  current_frame << shot
+  if current_frame[0] == 10 || current_frame.size == 2
+    frames << current_frame
+    current_frame = []
+  end
+end
+
+frames << current_frame unless current_frame.empty?
```

Comment:
```suggestion
current_frame = []
frames = [current_frame]

shots.each do |shot|
  current_frame << shot
  if current_frame[0] == 10 || current_frame.size == 2
    current_frame = []
    frames << current_frame
  end
end
```

ちょっとややこしいかもしれませんが、参照概念を理解していると上のようなコードでも同じように動くはずです。
ぱっと理解できない場合はじっくり考えてみてください。